gulpfile.ts

```diff
@@ -0,0 +1,225 @@
+      parserOptions: {
+        requireConfigFile: false
+      },
+      plugins: ['unused-imports'],
+      rules: {
+        'unused-imports/no-unused-imports': 'error'
+      }
+    },
+    fix: true,
+    ignore: false,
+    useEslintrc: false
+  });
+}
+
+function babelStream() {
+  return babel({
+    plugins: ['@babel/plugin-transform-modules-commonjs']
+  });
+}
+
+function filterMacros() {
+  return filter(['**/*.ts', '**/*.js', '!**/*.macro.d.ts', '!**/*.macro.js']);
+}
+
+function gulpFactory(
+  input: {
+    globs: string | string[],
+    opts?: SrcOptions
+  },
+  output: string,
+  overrideSettings?: Omit<ts.Settings, 'outDir'>,
+  useCjsTransform = false) {
+  macros.clear();
+  const tsProject = createProject({
+    ...overrideSettings,
+    outDir: output
+  });
+  const cleanStream = gulp.src(output, { read: false, allowEmpty: true })
+    .pipe(clean());
+  const tsBuildResult = gulp.src(input.globs, input.opts)
+    .pipe(tsProject());
+  const jsBuildResult = tsBuildResult.js
+    .pipe(eslintStream())
+    .pipe(eslint.fix())
+    .pipe(eslint.format())
+    .pipe(eslint.failAfterError())
+    .pipe(useCjsTransform ? babelStream() : tap(() => { /* */ }))
+    .pipe(terserStream());
+  return merge(
+    cleanStream,
+    merge([tsBuildResult.dts, jsBuildResult])
+      .pipe(filterMacros())
+      .pipe(gulp.dest(output)));
+}
+
+gulp.task(
+  'cjs',
+  () => gulpFactory(
+    { globs: 'src/**/*.ts' },
+    'dist/cjs',
+    {
+      module: 'ES2015',
+      declaration: true
+    },
+    true)
+);
+
+gulp.task(
+  'esm',
+  () => gulpFactory(
+    { globs: 'src/**/*.ts' },
+    'dist/esm',
+    {
+      target: 'ES5',
+      module: 'ES2015',
+      declaration: true
+    })
+);
+
+gulp.task(
+  'performanceTest',
+  () => gulpFactory(
+    {
+      globs: [
+        'src/**/*.ts',
+        'performance/**/*.ts'
+      ],
+      opts: {
+        base: '.'
+      }
+    },
+    'dist/performance',
+    {
+      module: 'ES2015',
+      declaration: false
+    },
+    true)
+);
+
+function gulpUmdFactory(
+  input: string,
+  output: string,
+  uglifyOpts?: GulpUglify.Options,
+  sourcemap = false
+) {
+  macros.clear();
+  return gulp.src(`dist/umd/${output}`, { read: false, allowEmpty: true })
+    .pipe(clean())
+    .pipe(gulp.src(input))
+    .pipe(sourcemap ? sourcemaps.init() : tap(() => { /* */ }))
+    .pipe(rollupStream(output))
+    .pipe(terserStream())
+    .pipe(GulpUglify(uglifyOpts))
+    .pipe(sourcemap ? sourcemaps.write('.') : tap(() => { /* */ }))
+    .pipe(gulp.dest('dist/umd'));
+}
+
+gulp.task(
+  'umd',
+  () => gulpUmdFactory(
+    'src/**/*.ts',
+    'js-sdsl.js',
+    {
+      compress: false,
+      output: { beautify: true },
+      mangle: { keep_fnames: true }
+    }
+  )
+);
+
+gulp.task(
+  'umd:min',
+  () => gulpUmdFactory(
+    'src/**/*.ts',
+    'js-sdsl.min.js',
+    {
+      compress: true
+    },
+    true
+  )
+);
+
+gulp.task('default', gulp.series('cjs', 'esm', 'umd:min'));
```

Comment:
Change here to `gulp.task('default', gulp.series('cjs', 'esm', 'umd', 'umd:min'));`